logdna/request.go

```diff
@@ -0,0 +1,139 @@
+package logdna
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type httpRequest func(string, string, io.Reader) (*http.Request, error)
+type bodyReader func(io.Reader) ([]byte, error)
+type jsonMarshal func(interface{}) ([]byte, error)
+type httpClientInterface interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
+// Client used to make HTTP requests to the configuration api
```

Comment:
nit: the comment for `newRequestConfig` better describes the purpose of this wrapper struct

Reply:
Yeah, good call.  *All* of these comments need updating.  They mostly just repeat the obvious.

Reply:
done